adapter/src/main/java/com/twilio/kudu/sql/schema/KuduSchema.java

```diff
@@ -153,26 +180,93 @@ protected Map<String, Table> getTableMap() {
       KuduTableMetadata kuduTableMetadata = entry.getValue();
       final List<String> descendingOrderedColumnNames = kuduTableMetadata.getDescendingOrderedColumnNames();
 
+      String factTableName = entry.getKey();
       // create any cube tables first
       List<CalciteKuduTable> cubeTableList = new ArrayList<>(5);
+      String timestampColumnName = kuduTableMetadata.getTimestampColumnName();
       for (CubeTableInfo cubeTableInfo : kuduTableMetadata.getCubeTableInfo()) {
         Optional<KuduTable> cubeTableOptional = openKuduTable(cubeTableInfo.tableName);
-        cubeTableOptional.ifPresent(kuduTable -> {
-          final CalciteKuduTableBuilder builder = new CalciteKuduTableBuilder(kuduTable, client)
+        cubeTableOptional.ifPresent(cubeKuduTable -> {
+          final CalciteKuduTableBuilder builder = new CalciteKuduTableBuilder(cubeKuduTable, client)
               .setEnableInserts(enableInserts).setDisableCubeAggregation(disableCubeAggregation)
               .setReadSnapshotTimeDifference(readSnapshotTimeDifference)
               .setTableType(com.twilio.kudu.sql.TableType.CUBE)
               .setEventTimeAggregationType(cubeTableInfo.eventTimeAggregationType);
-          setDescendingFieldIndices(builder, descendingOrderedColumnNames, kuduTable);
-          setTimestampColumnIndex(builder, kuduTableMetadata.getTimestampColumnName(), kuduTable);
+          setDescendingFieldIndices(builder, descendingOrderedColumnNames, cubeKuduTable);
+          setTimestampColumnIndex(builder, kuduTableMetadata.getTimestampColumnName(), cubeKuduTable);
           CalciteKuduTable calciteKuduTable = builder.build();
           tableMap.put(cubeTableInfo.tableName, calciteKuduTable);
           cubeTableList.add(calciteKuduTable);
+
+          StringBuilder queryBuilder = new StringBuilder("SELECT ");
```

Comment:
Using the cube schema generate a view sql of the form SELECT aggregate function FROM table GROUP BY ...